source/server/configuration_impl.cc

```diff
@@ -14,13 +14,14 @@
 namespace Server {
 namespace Configuration {
 
-void FilterChainUtility::buildFilterChain(Network::FilterManager& filter_manager,
+bool FilterChainUtility::buildFilterChain(Network::FilterManager& filter_manager,
                                           const std::list<NetworkFilterFactoryCb>& factories) {
   for (const NetworkFilterFactoryCb& factory : factories) {
     factory(filter_manager);
   }
 
   filter_manager.initializeReadFilters();
+  return factories.size() > 0;
```

Comment:
Thanks for the suggestions. FilterManager::initializeReadFilter() already had an ASSERT for non-empty upstream filters. I replaced the build-time check with a runtime check and propagated that error back so the connection can be closed if filter initialization fails.